### PR TITLE
Timezone support

### PR DESCRIPTION
**Pull request by Jaskar14**
@santigimeno Making a new PR as requested, and merged in the most recent version of master into the branch. I'll start modifying this branch with the comments [here](https://github.com/node-schedule/node-schedule/pull/316) as suggested. 
Can you also do me a favor, and assign yourself as a reviewer and myself as a assignee?

### CODE REVIEW

**Comment by santigimeno** [comment]
@Jaskar14, thanks for doing this! Before I look into this, the tests are failing, can you look into that?
Thanks!

**Comment by Jaskar14** [comment]
@santigimeno yep, going to look into that tonight. got caught up in a few things.

**Comment by Jaskar14** [comment]
@santigimeno Not sure why, but two of the tests are failing. They're both in the schedule-cron-jobs, the last two tests. Both are passing on my local machine, but not on travis-ci. Can you provide any insight as to what may be causing the issue?
I'll take a look into this over the weekend if you're able to provide some feedback

File: test/date-convenience-methods-test.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ module.exports = {
       }, 1250);
 
       clock.tick(1250);
-    },
+    },
```
Comment (santigimeno) [inline test/date-convenience-methods-test.js]: This change is not needed

File: test/schedule-cron-jobs.js [hunk 2]
```diff
@@ -126,6 +126,38 @@ module.exports = {
 
       clock.tick(timeout);
     },
+    "Runs job every year in Vancouver": function(test) {
+      test.expect(4);
+
+      var timeout = 4 * 365.25 * 24 * 60 * 60 * 1000;
```
Comment (santigimeno) [inline test/schedule-cron-jobs.js]: Probably you could increase this a little to be sure it runs 4 times

File: test/schedule-cron-jobs.js [hunk 3]
```diff
@@ -126,6 +126,38 @@ module.exports = {
...
+        job.cancel();
+        test.done();
+      }, timeout);
+
+      clock.tick(timeout);
+    },
+    "Runs job every year in Kuala Lumpur": function(test) {
+      test.expect(4);
+
+      var timeout = 4 * 365.25 * 24 * 60 * 60 * 1000;
```
Comment (santigimeno) [inline test/schedule-cron-jobs.js]: same here

**Comment by santigimeno** [comment]
@Jaskar14 sorry for the delay. By looking at the PR it seems there are only changes on the tests but not on the `lib/schedule.js`. Is this intended or are the changes missing?

Also, a couple of things:
- Could you rebase to current master? 
- Could you remove `package-lock.json` file?

WRT the tests, TBH, I don't know what's going on as the tests also pass in my local machine. You could update this branch with some logs to see what's going on travis.

Thanks

**Comment by stewshka** [comment]
updates on this?

**Comment by SilencerWeb** [comment]
any updates?

@santigimeno

**Comment by Freakspot** [comment]
@santigimeno friendly reminder

**Comment by Exitare** [comment]
Any updates on this ?

**Comment by jasonhargrove** [comment]
Nice

**Comment by kibertoad** [comment]
@Jaskar14 Do you intend to ever implement the actual change?

**Comment by Exitare** [comment]
Certainly wouldn't count on it.

**Comment by Jaskar14** [comment]
@kibertoad sorry, I lost track of this a while ago. I'm going to close this pr and if someone else wants to wrap it up, be my guest. 
Apologies

**Pull request closed without merging**